File: DynamicProgramming/FindTraining/train.py
class Solution:
    def findWinner(self, n, x, y):
        dp = [False] * (n + 1)

        for i in range(1, n + 1):
            a = dp[i - 1] if i - 1 >= 0 else True
            b = dp[i - x] if i - x >= 0 else True
            c = dp[i - y] if i - y >= 0 else True

            dp[i] = not (a and b and c)

        return dp[n]

File: DynamicProgramming/FindTraining/train.cpp
#include <vector>
using namespace std;

class Solution {
  public:
    int findWinner(int n, int x, int y) {
        // code here
        vector<bool> dp(n + 1, false);
        for(int i = 1; i <= n; i++){
            bool a = (i-1 >= 0) ? dp[i-1] : true;
            bool b = (i-x >= 0) ? dp[i-x] : true;
            bool c = (i-y >= 0) ? dp[i-y] : true;
        
            dp[i] = !(a && b && c);
        }
        return dp[n];
    }
};
